File: leetcode/array_string/daily2788.dart
void main(List<String> args) {
  var solution = Solution();

  // case1
  print(solution
      .splitWordsBySeparator(["one.two.three", "four.five", "six"], '.'));
  // case2
  print(solution.splitWordsBySeparator(["\$easy\$", "\$problem\$"], '\$'));
  // case3s
  print(solution.splitWordsBySeparator(["|||"], '|'));
}

class Solution {
  List<String> splitWordsBySeparator(List<String> words, String separator) {
    List<String> ans = [];
    for (var word in words) {
      List<String> split = word.split(separator);
      for (var s in split) {
        if (s.isNotEmpty && s != separator) {
          ans.add(s);
        }
      }
    }

    return ans;
  }
}

File: leetcode/arraystring/daily2788.java
package arraystring;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;
import java.util.Objects;

public class daily2788 {
    public static void main(String[] args) {
        Solution2788 solution = new Solution2788();

        // case1
        System.out.println(solution.splitWordsBySeparator(Arrays.asList("one.two.three", "four.five", "six"), '.'));
        // case2
        System.out.println(solution.splitWordsBySeparator(Arrays.asList("$easy$", "$problem$"), '$'));
        // case3
        System.out.println(solution.splitWordsBySeparator(Arrays.asList("|||"), '|'));
    }
}

class Solution2788 {
    public List<String> splitWordsBySeparator(List<String> words, char separator) {
        ArrayList<String> ans = new ArrayList<String>();
        for (String word : words) {
            String[] split = word.split("\\" + separator);
            if (split.length != 0) {
                for (String s : split) {
                    if (!Objects.equals(s, "") && !Objects.equals(s, String.valueOf(separator))) {
                        ans.add(s);
                    }
                }
            }
        }

        return ans;
    }
}
